astropy/cosmology/_io/cosmology.py

```diff
@@ -84,9 +99,14 @@ def to_cosmology(cosmo, *args):
     return cosmo
 
 
-def cosmology_identify(origin, format, *args, **kwargs):
+def cosmology_identify(
+    origin: str, format: str | None, *args: Any, **kwargs: object
```

Comment:
Why is the annotation for `format` so specific? From the code it is clear that `format` could be any `object` and the code would still work. There's a couple more similar cases in the patch (mostly involving `Literal`).

Reply:
> From the code it is clear that format could be any object and the code would still work

While true, `None` and `str` are the only types that actually make sense here, from a user perspective (I think ?). The fact that there are no actual type enforcement at runtime doesn't seem relevant to me: it's an implementation detail, which shouldn't justify making annotations less useful.

Reply:
To be honest I have trouble understanding why `format` exists. If I want to know if something is identifiable as a `Cosmology` then simply calling `cosmology_identify()` should be sufficient and having the option of specifying `format="astropy.cosmology"` doesn't seem to have any purpose.

Reply:
I thought it was a requirement on the signature of functions that are passed to `_UnifiedIORegistryBase.register_identifier` but it doesn't actually look to be the case... I don't get it either.

Reply:
The I/O in Cosmology was based on the implementation from `table`.

Reply:
Could you point out the specific lines of code?

Reply:
But none of these have a `format` parameter.

Reply:
They are the same as "filepath". But here we are not writing, so it's "format". E.g. "parquet_identify" does a check on `filepath` to determine it ends in `". parquet"` or `"parq"

Reply:
[For type checking purposes a function argument annotation should be as general as possible](https://peps.python.org/pep-0484/#covariance-and-contravariance). A more specific annotation can be justified if it is simpler but still helpful to users. In this case `format: object` would be more useful for type checking than `format: str | None` and the latter can't be justified by its helpfulness because I can't think of any use cases where I'd want to specify `format`, meaning that the entire argument is useless regardless of how it is annotated.